arviz/stats/ecdf_utils.py

```diff
@@ -129,6 +137,90 @@ def ecdf_confidence_band(
     return prob_lower, prob_upper
 
 
+def _update_ecdf_band_interior_probabilities(
+    prob_left: np.ndarray,
+    interval_left: np.ndarray,
+    interval_right: np.ndarray,
+    cdf_left: float,
+    cdf_right: float,
+    ndraws: int,
+) -> np.ndarray:
+    """Update the probability that an ECDF has been within the envelope including at the current
+    point.
+
+    Arguments
+    ---------
+    prob_left : np.ndarray
+        For each point in the interior at the previous point, the joint probability that it and all
+        points before are in the interior.
+    interval_left : np.ndarray
+        The set of points in the interior at the previous point.
+    interval_right : np.ndarray
+        The set of points in the interior at the current point.
+    cdf_left : float
+        The CDF at the previous point.
+    cdf_right : float
+        The CDF at the current point.
+    ndraws : int
+        Number of draws in the original dataset.
+
+    Returns
+    -------
+    prob_right : np.ndarray
+        For each point in the interior at the current point, the joint probability that it and all
+        previous points are in the interior.
+    """
+    z_tilde = (cdf_right - cdf_left) / (1 - cdf_left)
+    interval_left = interval_left[:, np.newaxis]
+    prob_conditional = binom.pmf(interval_right, ndraws - interval_left, z_tilde, loc=interval_left)
+    prob_right = prob_left.dot(prob_conditional)
+    return prob_right
+
+
+def _ecdf_band_optimization_objective(
```

Comment:
This should ideally be jitted with numba if available for a likely significant speed-up. Currently the problem is that the functions it calls use `scipy.stats.binom.interval` and `scipy.stats.binom.pmf`, which are not JIT-able. If we reimplement these functions here, then we could probably get a significant speed-up.

Reply:
`interval` calls `ppf`, which defers to Boost's implementation, which in turn uses TOMS Algorithm 748 to find a root in an interval: https://doi.org/10.1145/210089.210111, so reimplementing that to be JIT-able would be the bottleneck. Which seems like it could be a lot of work.

Reply:
Ah wait, since `interval` is called outside the for loop and only `pmf` is called within, this could be simple.

`@conditional_jit` doesn't seem to like it if I call another conditionally-jitted function within the first conditionally-jitted function. Is there a way to make this work?